frontend/src/extensions.ts

```diff
@@ -9,6 +9,54 @@ import { getUrlPath, urlFor } from "./helpers";
 import { log_error } from "./log";
 import { extensions } from "./stores";
 
+class ExtensionApi {
+  extension_name: string;
+
+  constructor(extension_name: string) {
+    this.extension_name = extension_name;
+  }
+
+  request(
+    endpoint: string,
+    method: string,
+    params: Record<string, string | number> | undefined,
+    body: object | undefined
+  ) {
+    const url = urlFor(
+      `extension/${this.extension_name}/${endpoint}`,
+      params,
+      false
+    );
+    let opts = {};
+    if (body) {
+      opts =
+        body instanceof FormData
+          ? { body }
+          : {
+              headers: { "Content-Type": "application/json" },
+              body: JSON.stringify(body),
+            };
+    }
+    return fetch(url, { method, ...opts }).then((res) => res.json());
```

Comment:
This should import fetch from lib/fetch to ensure that cookies are passed for example.

Reply:
Done